src/test/kotlin/com/example/probation/controller/UsersControllerTest.kt

```diff
@@ -0,0 +1,115 @@
+package com.example.probation.controller
+
+import com.example.probation.ProbationApplication
+import com.example.probation.core.entity.Role
+import com.example.probation.repository.UsersRepository
+import com.example.probation.service.impl.UsersServiceImpl
+import com.jayway.jsonpath.JsonPath
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.security.test.context.support.WithUserDetails
+import org.springframework.security.test.web.servlet.response.SecurityMockMvcResultMatchers
+import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.jdbc.Sql
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import org.springframework.test.web.servlet.setup.DefaultMockMvcBuilder
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import org.springframework.web.context.WebApplicationContext
+import java.util.*
+import java.util.stream.Collectors
+
+@ExtendWith(SpringExtension::class)
+@AutoConfigureMockMvc
+@Sql("/init.sql")
+@ActiveProfiles("local")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
```

Comment:
also in such cases, it is always better to have some abstract common class with the configuration annotations set and extend this class by the test classes